File: 674. Longest Continuous Increasing Subsequence/674.go
package main

import "fmt"

func main(){
	fmt.Println(findLengthOfLCIS([]int{1,1,1,1,1}))
}

func findLengthOfLCIS(nums []int) int {
	max_incerasing_length, tmp_max := 0, 1
	if len(nums) == 0{
		return max_incerasing_length
	}
	for i:=0; i<len(nums)-1; i++{
		if nums[i] < nums[i+1]{
			tmp_max ++
		}else{
			if max_incerasing_length<tmp_max{
				max_incerasing_length = tmp_max
			}
			tmp_max = 1
		}
	}
	if max_incerasing_length<tmp_max{
		max_incerasing_length = tmp_max
	}
	return max_incerasing_length
}

File: 674. Longest Continuous Increasing Subsequence/674.py
class Solution:
    def findLengthOfLCIS(self, nums) -> int:
        tmp = 1
        fin = 1
        if not nums:
            return 0
        for index in range(1,len(nums)):
            if nums[index]>nums[index-1]:
                tmp += 1
            else:
                if fin < tmp :
                    fin = tmp
                tmp = 1
        if fin < tmp :
            fin = tmp
        return fin
    
    # DP的思想
    def findLengthOfLCIS1(self, nums):
        if not nums:
            return 0
        dp = [1] * len(nums)
        for i in range(1, len(nums)):
            if nums[i] > nums[i-1]:
                dp[i] = dp[i - 1] + 1
        return max(dp)

a = Solution()
print(a.findLengthOfLCIS([-1]))